File: Programmers/커뮤러닝/1주차-그리디,정렬,이분탐색,시뮬레이션/가장큰수.java
/*  Tip. Java API 활용
    1. Java 언어를 사용한다 == Java가 제공하는 라이브러리를 사용
    2. 기본 라이브러리인 java.lang.*, java.util.*의 사용법을 숙지  */

import java.util.*;

class Solution {
    public String solution(int[] numbers) {

        // Java 8의 람다식을 활용한 풀이
        String answer = IntStream.of(numbers).mapToObj(String::valueOf).sorted((s1,s2) -> (s2+s1).compareTo(s1+s2)).collect(Collectors.joining());
        
        /*
        String answer = "";
        String[] arr = new String[numbers.length];
        
        for (int i = 0; i < numbers.length; ++i)
            arr[i] = String.valueOf(numbers[i]);

        Arrays.sort(arr, new Comparator<String>() {
            @Override
            public int compare(String a, String b) {
                return (b+a).compareTo(a+b);
            }
        });
        
        for (String str : arr)
            answer += str;
        */

        if (arr[0].charAt(0) == '0') return "0";

        return answer;
    }
}

File: Programmers/LEVEL 2/가장큰수.py
import functools


def comparator(x, y):
    str1 = x + y
    str2 = y + x

    if str1 < str2:
        return -1
    else:
        return 1


def solution(numbers):
    data = list(map(str, numbers))
    data.sort(key=functools.cmp_to_key(comparator), reverse=True)
    answer = str(int(''.join(data)))
    return answer
